vault/resource_nomad_secret_backend.go

```diff
@@ -0,0 +1,304 @@
+		data["client_key"] = v
+	}
+
+	if v, ok := d.GetOkExists("max_token_name_length"); ok {
+		data["max_token_name_length"] = v
+	}
+
+	if v, ok := d.GetOkExists("token"); ok {
+		data["token"] = v
+	}
+
+	configPath := fmt.Sprintf("%s/config/access", backend)
+	log.Printf("[DEBUG] Writing %q", configPath)
+	if _, err := client.Logical().Write(configPath, data); err != nil {
+		return fmt.Errorf("error writing %q: %s", configPath, err)
+	}
+
+	log.Printf("[DEBUG] Wrote %q", configPath)
+	return readNomadAccessConfigResource(d, meta)
+}
+
+func readNomadAccessConfigResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+	log.Printf("[DEBUG] Reading %q", path)
+
+	mountResp, err := client.Sys().MountConfig(path)
+	if err != nil && util.Is404(err) {
+		log.Printf("[WARN] %q not found, removing from state", path)
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("error reading %q: %s", path, err)
+	}
+
+	d.Set("backend", d.Id())
+	d.Set("default_lease_ttl_seconds", mountResp.DefaultLeaseTTL)
+	d.Set("max_lease_ttl_seconds", mountResp.MaxLeaseTTL)
+
+	configPath := fmt.Sprintf("%s/config/access", d.Id())
+	log.Printf("[DEBUG] Reading %q", configPath)
+
+	resp, err := client.Logical().Read(configPath)
+	if err != nil {
+		return fmt.Errorf("error reading %q: %s", configPath, err)
+	}
+	log.Printf("[DEBUG] Read %q", configPath)
+	if resp == nil {
+		log.Printf("[WARN] %q not found, removing from state", configPath)
+		d.SetId("")
+		return nil
+	}
+
+	if val, ok := resp.Data["address"]; ok {
+		if err := d.Set("address", val); err != nil {
+			return fmt.Errorf("error setting state key 'address': %s", err)
+		}
+	}
+
+	if val, ok := resp.Data["ca_cert"]; ok {
+		if err := d.Set("ca_cert", val); err != nil {
+			return fmt.Errorf("error setting state key 'ca_cert': %s", err)
+		}
+	}
+
+	if val, ok := resp.Data["client_cert"]; ok {
+		if err := d.Set("client_cert", val); err != nil {
+			return fmt.Errorf("error setting state key 'client_cert': %s", err)
+		}
+	}
+
+	if val, ok := resp.Data["client_key"]; ok {
+		if err := d.Set("client_key", val); err != nil {
+			return fmt.Errorf("error setting state key 'client_key': %s", err)
+		}
+	}
+
+	if val, ok := resp.Data["max_token_name_length"]; ok {
+		if err := d.Set("max_token_name_length", val); err != nil {
+			return fmt.Errorf("error setting state key 'max_token_name_length': %s", err)
+		}
+	}
+
+	if val, ok := resp.Data["token"]; ok {
+		if err := d.Set("token", val); err != nil {
+			return fmt.Errorf("error setting state key 'token': %s", err)
+		}
+	}
+
+	return nil
+}
+
+func updateNomadAccessConfigResource(d *schema.ResourceData, meta interface{}) error {
+	backend := d.Id()
+
+	client := meta.(*api.Client)
+	defaultTTL := d.Get("default_lease_ttl_seconds").(int)
+	maxTTL := d.Get("max_lease_ttl_seconds").(int)
+	tune := api.MountConfigInput{}
+	data := map[string]interface{}{}
+
+	if defaultTTL != 0 {
```

Comment:
I believe this prevents us from initially setting a default lease to something that is not the default system lease, and then removing it, e.g. we can't set it to something non-standard and then try to later adapt the system defaults implicitly by removing our non-standard value. 

Example:

- System default is `5m`
- We use `default_lease_ttl_seconds` to set it to `6m` instead
- Later we decide we want to use the system defaults, so we remove our `default_lease_ttl_seconds`
- No change is detected here, because the new value would be the zero value, right? 
- No update happens, and the default least ttl remains at `6m`

Is my assumption correct here?

Reply:
Agreed, fixed.